auth/api/iam/openid4vp.go

```diff
@@ -97,6 +99,19 @@ func (r Wrapper) handleAuthorizeRequestFromHolder(ctx context.Context, verifier
 	if params.get(oauth.NonceParam) == "" {
 		return nil, withCallbackURI(oauthError(oauth.InvalidRequest, "missing nonce parameter"), redirectURL)
 	}
+	// we require PKCE (RFC7636) for authorization code flows
+	// check code_challenge and code_challenge_method
+	if params.get(oauth.CodeChallengeParam) == "" || params.get(oauth.CodeChallengeMethodParam) == "" {
+		return nil, oauthError(oauth.InvalidRequest, "missing code_challenge or code_challenge_method parameter")
+	}
+	// for PKCE we also need the state parameter
+	if params.get(oauth.StateParam) == "" {
+		return nil, oauthError(oauth.InvalidRequest, "missing state parameter")
+	}
+	serverSession := createSession(params, verifier)
+	if err := r.oauthServerStateStore().Put(serverSession.ClientState, serverSession); err != nil {
+		return nil, oauthError(oauth.ServerError, "failed to store server state")
+	}
```

Comment:
if PKCE is the only reason to start a server session, we could consider encrypting the params into the authorization as suggested by the [RFC](https://datatracker.ietf.org/doc/html/rfc7636#section-4.4)

> Typically, the "code_challenge" and "code_challenge_method" values are stored in encrypted form in the "code" itself but could alternatively be stored on the server associated with the code.